driver-core/src/main/com/mongodb/client/model/expressions/BooleanExpression.java

```diff
@@ -19,49 +19,47 @@
 import java.util.function.Function;
 
 /**
- * Expresses a boolean value.
+ * A logical boolean value, either true or false.
  */
 public interface BooleanExpression extends Expression {
 
     /**
-     * Returns logical true if this expression evaluates to logical false.
-     * Returns logical false if this expression evaluates to logical true.
+     * The logical negation of {@code this} value.
      *
-     * @return True if false; false if true.
+     * @return the resulting value.
      */
     BooleanExpression not();
 
     /**
-     * Returns logical true if this or the other expression evaluates to logical
-     * true. Returns logical false if both evaluate to logical false.
+     * The logical conjunction of {@code this} and the {@code other} value.
      *
-     * @param or the other boolean expression.
-     * @return True if either true, false if both false.
+     * @param other the other boolean value.
+     * @return the resulting value.
      */
-    BooleanExpression or(BooleanExpression or);
+    BooleanExpression or(BooleanExpression other);
 
     /**
-     * Returns logical true if both this and the other expression evaluate to
-     * logical true. Returns logical false if either evaluate to logical false.
+     * The logical disjunction of {@code this} and the {@code other} value.
      *
-     * @param and the other boolean expression.
-     * @return true if both true, false if either false.
+     * @param other the other boolean value.
+     * @return the resulting value.
      */
-    BooleanExpression and(BooleanExpression and);
+    BooleanExpression and(BooleanExpression other);
+    // TODO-END check the evaluation semantics of and/or
 
     /**
-     * If this expression evaluates to logical true, returns the result of the
-     * evaluated left branch expression. If this evaluates to logical false,
-     * returns the result of the evaluated right branch expression.
+     * The {@code left} branch when {@code this} is true,
+     * and the {@code right} branch otherwise.
      *
-     * @param left the left branch expression
-     * @param right the right branch expression
-     * @return left if true, right if false.
+     * @param left the left branch.
+     * @param right the right branch.
+     * @return the resulting value.
```

Comment:
Right, this focused too much on evaluation semantics.

Reply:
It still uses "branch" in the description of the method.

Reply:
I should have caught that.

By the way, do you see another way to implement cond? It has been mentioned a few times in the design document, that users would find it strange to chain `cond` off of a boolean in this way. The alternative I can imagine is to have a "nested functional" cond static method that takes 3 params, but this seems inconsistent with the design approach. It is also possible to have `bigExpressionHere.cond(e -> e.eq(1), e -> leftExpression, e -> rightExpression)` (which is a bit more similar to our `switch`), but it seemed very verbose (and I vaguely remember there was a more technical problem with it, which I cannot now recall).

Reply:
> cond static method that takes 3 params, but this seems inconsistent with the design approach

Yes, everything else is chainable, and a static method is not.

> bigExpressionHere.cond(e -> e.eq(1), e -> leftExpression, e -> rightExpression)

This seems fine at first glance, but is indeed similar to
```java
of(0).switchOn(on -> on.eq(of(0), v -> of("A")).defaults(v -> of("B")))
```
which makes this form almost useless. I think it's fine for us to have `cond` as it is now, and if users don't like it, maybe they will let us know.

P.S. For some reason the code I wrote doesn't compile, but I don't immediately see what the problem is. I will leave a comment in https://github.com/mongodb/mongo-java-driver/pull/1055, because this seems worth looking at.